File: AndroidProject/CollectorAgentMind/src/main/java/com/example/collectoragentmind/codelets/behaviors/GoToClosestApple.java
package com.example.collectoragentmind.codelets.behaviors;

import com.example.collectoragentmind.util.Apple;

import br.unicamp.cst.core.entities.Codelet;
import br.unicamp.cst.core.entities.MemoryObject;

public class GoToClosestApple extends Codelet {

    /*
        If it has a closest apple, sends a command to
        the legsMO with its position
    */

    private MemoryObject closestAppleMO;
    private MemoryObject legsMO;

    @Override
    public void accessMemoryObjects() {
        closestAppleMO = (MemoryObject) getInput("CLOSEST_APPLE");
        legsMO = (MemoryObject) getOutput("LEGS");
    }

    @Override
    public void calculateActivation() {

    }

    @Override
    public void proc() {
        Apple closestApple = (Apple) closestAppleMO.getI();

        if (closestApple != null) {
            legsMO.setI(closestApple.position);
        }
    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/cst/codelets/behaviors/GoToClosestApple.kt
package com.example.game.cst.codelets.behaviors

import br.unicamp.cst.core.entities.Codelet
import br.unicamp.cst.core.entities.MemoryObject
import com.example.game.environment.Apple

class GoToClosestApple : Codelet() {
    /*
        If it has a closest apple, sends a command to
        the legsMO with its position
    */

    private var closestAppleMO: MemoryObject? = null
    private var legsMO: MemoryObject? = null

    override fun accessMemoryObjects() {
        closestAppleMO = getInput("CLOSEST_APPLE") as MemoryObject
        legsMO = getOutput("LEGS") as MemoryObject
    }

    override fun calculateActivation() {}

    override fun proc() {
        val closestApple: Apple? = closestAppleMO?.i as Apple?
        if (closestApple != null) {
            legsMO!!.i = closestApple.getPos()
        }
    }
}